compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/BallerinaSemanticModel.java

```diff
@@ -123,9 +123,9 @@ public BallerinaSemanticModel(BLangPackage bLangPackage, CompilerContext context
         for (Map.Entry<Name, List<Scope.ScopeEntry>> entry : scopeSymbols.entrySet()) {
             Name name = entry.getKey();
             List<Scope.ScopeEntry> scopeEntries = entry.getValue();
-
             for (Scope.ScopeEntry scopeEntry : scopeEntries) {
-                addToCompiledSymbols(compiledSymbols, scopeEntry, cursorPos, name, statesSet);
+                boolean isChildOfCurrentNode = symbolEnv.scope.owner.equals(scopeEntry.symbol.owner);
```

Comment:
This may be problematic since we don't have an `equals()` implementation for symbols and our symbol implementation doesn't guarantee that the same symbol will always be in use.

Reply:
In that case, we need to perform different equivalency checks for the `pkgID`, `name`, and the `pos` of these owner symbols. 
That would derive that the two symbols are equal. 

@pubudu91, WDYT?

Reply:
Hmm yeah. Would have to try something like that I guess 🤔  Also, see if deriving this info using the available node info is a viable approach

Reply:
Updated with the above-mentioned approach.